pkg/service/catalog/api/types.go

```diff
@@ -0,0 +1,21 @@
+package api
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+type CatalogService struct {
+	// Target is a reference to a service
+	Target kapi.ObjectReference
+	// Type is a published service's claim type
+	Type string
+	// Description is a user-defined description of the published service
+	Description string
```

Comment:
We use annotations for descriptions on core objects.  A cli that shows the
core object and the virtual object has to filter out the annotation if we
have two fields (annotation and core).

A counter argument is that can a catalog service have a different
description than the core service and the answer is yes.  The project
argument was weaker because projects are 1:1 with namespaces.

On Oct 2, 2015, at 10:17 AM, Andy Goldstein <notifications@github.com>
wrote:

In pkg/service/catalog/api/types.go
<https://github.com/openshift/origin/pull/4822#discussion_r41025768>:

> @@ -0,0 +1,21 @@
> +package api
> +
> +import (
> +	kapi "k8s.io/kubernetes/pkg/api"
> +)
> +
> +type CatalogService struct {
> +	// Target is a reference to a service
> +	Target kapi.ObjectReference
> +	// Type is a published service's claim type
> +	Type string
> +	// Description is a user-defined description of the published service
> +	Description string

It just seems silly to use an annotation when we can do a first class field.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4822/files#r41025768>.